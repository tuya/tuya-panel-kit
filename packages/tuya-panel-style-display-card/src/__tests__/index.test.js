/**
 * @jest-environment jsdom
 */
 import React from 'react';
 import renderer from 'react-test-renderer';
 import TuyaRNSvgs from 'tuya-panel-kit/lib/components/iconfont/svg/defaultSvg';
 import { ClassicDisplayCard, AcrylicDisplayCard, NordicDisplayCard } from '../index';
 
 describe('DisplayCard', () => {
   it('ClassicDisplayCard', () => {
     const wrapper = renderer
       .create(<ClassicDisplayCard backgroundColor="#FFF" />)
       .toJSON();
     expect(wrapper).toMatchSnapshot();
   });
   it('NordicDisplayCard', () => {
    const wrapper = renderer
    .create(<NordicDisplayCard icon={TuyaRNSvgs.power} />)
    .toJSON();
  expect(wrapper).toMatchSnapshot();
   })
   it('AcrylicDisplayCard', () => {
     const wrapper = renderer.create(<AcrylicDisplayCard />).toJSON();
     expect(wrapper).toMatchSnapshot();
   });
 });
 