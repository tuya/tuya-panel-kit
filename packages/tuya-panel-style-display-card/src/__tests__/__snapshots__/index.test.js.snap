// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayCard AcrylicDisplayCard 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderRadius": 0,
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingBottom": 52,
        "paddingLeft": 62,
        "paddingRight": 38,
        "paddingTop": 52,
        "width": 396,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flexDirection": "row",
        }
      }
    >
      <Text
        accessible={true}
        align={null}
        allowFontScaling={false}
        color="#000"
        ellipsizeMode="tail"
        size={128}
        style={
          Array [
            false,
            Array [
              false,
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#000",
                  "fontSize": 128,
                  "fontWeight": "600",
                },
                Array [
                  Object {
                    "lineHeight": 180,
                  },
                  Object {
                    "lineHeight": 175,
                  },
                ],
              ],
            ],
          ]
        }
        text="32"
        type={null}
        weight={600}
      >
        32
      </Text>
      <Text
        accessible={true}
        align={null}
        allowFontScaling={false}
        color="rgba(0, 0, 0, 0.5)"
        ellipsizeMode="tail"
        size={36}
        style={
          Array [
            false,
            Array [
              false,
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "rgba(0, 0, 0, 0.5)",
                  "fontSize": 36,
                  "fontWeight": "normal",
                },
                Array [
                  Object {
                    "lineHeight": 48,
                    "marginLeft": 10,
                    "marginTop": 20,
                  },
                  Object {
                    "lineHeight": 50,
                    "paddingBottom": 28,
                  },
                ],
              ],
            ],
          ]
        }
        text="째C"
        type={null}
        weight="normal"
      >
        째C
      </Text>
    </View>
    <Text
      accessible={true}
      align={null}
      allowFontScaling={false}
      color="rgba(0, 0, 0, 0.5)"
      ellipsizeMode="tail"
      size={28}
      style={
        Array [
          false,
          Array [
            false,
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "rgba(0, 0, 0, 0.5)",
                "fontSize": 28,
                "fontWeight": "500",
              },
              Array [
                Object {
                  "lineHeight": 40,
                  "marginTop": 8,
                },
                undefined,
              ],
            ],
          ],
        ]
      }
      text="Current Temp"
      type={null}
      weight={500}
    >
      Current Temp
    </Text>
  </View>
</View>
`;

exports[`DisplayCard ClassicDisplayCard 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFF",
        "borderRadius": 0,
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingBottom": 52,
        "paddingLeft": 62,
        "paddingRight": 38,
        "paddingTop": 52,
        "width": 396,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flexDirection": "row",
        }
      }
    >
      <Text
        accessible={true}
        align={null}
        allowFontScaling={false}
        color="#000"
        ellipsizeMode="tail"
        size={148}
        style={
          Array [
            false,
            Array [
              false,
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#000",
                  "fontSize": 148,
                  "fontWeight": "700",
                },
                Array [
                  Object {
                    "lineHeight": 180,
                  },
                  undefined,
                ],
              ],
            ],
          ]
        }
        text="32"
        type={null}
        weight={700}
      >
        32
      </Text>
      <Text
        accessible={true}
        align={null}
        allowFontScaling={false}
        color="#000"
        ellipsizeMode="tail"
        size={40}
        style={
          Array [
            false,
            Array [
              false,
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#000",
                  "fontSize": 40,
                  "fontWeight": "500",
                },
                Array [
                  Object {
                    "lineHeight": 48,
                    "marginLeft": 10,
                    "marginTop": 20,
                  },
                  undefined,
                ],
              ],
            ],
          ]
        }
        text="째C"
        type={null}
        weight={500}
      >
        째C
      </Text>
    </View>
    <Text
      accessible={true}
      align={null}
      allowFontScaling={false}
      color="rgba(0, 0, 0, 0.5)"
      ellipsizeMode="tail"
      size={28}
      style={
        Array [
          false,
          Array [
            false,
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "rgba(0, 0, 0, 0.5)",
                "fontSize": 28,
                "fontWeight": "400",
              },
              Array [
                Object {
                  "lineHeight": 40,
                  "marginTop": 8,
                },
                undefined,
              ],
            ],
          ],
        ]
      }
      text="Current Temp"
      type={null}
      weight={400}
    >
      Current Temp
    </Text>
  </View>
</View>
`;

exports[`DisplayCard NordicDisplayCard 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 24,
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingBottom": 40,
        "paddingLeft": 40,
        "paddingRight": 16,
        "paddingTop": 40,
        "width": 366,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#158CFB",
          "height": 50,
          "width": 50,
        },
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 50,
            "display": "flex",
            "height": 50,
            "justifyContent": "center",
            "overflow": "hidden",
            "width": 50,
          },
          Array [
            Object {
              "marginRight": 24,
            },
            Object {
              "marginRight": 40,
            },
          ],
        ],
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          Object {
            "height": 44,
            "width": 44,
          },
          false,
          null,
        ]
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            undefined,
            false,
            Object {
              "flex": 0,
              "height": 44,
              "width": 44,
            },
          ]
        }
        vbHeight={1024}
        vbWidth={1024}
      >
        <RNSVGPath
          d="M512 0C794.760533 0 1024 229.239467 1024 512S794.760533 1024 512 1024 0 794.760533 0 512 229.239467 0 512 0z m185.048178 327.0656a26.988089 26.988089 0 0 0-38.183822 38.183822 207.712711 207.712711 0 1 1-293.728712 0 26.988089 26.988089 0 1 0-38.183822-38.183822c-102.172444 102.1952-102.172444 267.901156 0 370.096356 102.1952 102.172444 267.901156 102.172444 370.096356 0 102.172444-102.1952 102.172444-267.901156 0-370.096356zM511.886222 227.555556a27.079111 27.079111 0 0 0-26.919822 24.302933l-0.136533 2.776178v196.152889a27.079111 27.079111 0 0 0 53.998933 2.776177l0.136533-2.776177v-196.152889a27.079111 27.079111 0 0 0-27.079111-27.079111z"
          fill={
            Array [
              0,
              1,
              1,
              1,
              1,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          propList={
            Array [
              "stroke",
              "strokeWidth",
              "fill",
            ]
          }
          stroke={
            Array [
              0,
              1,
              1,
              1,
              1,
            ]
          }
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
        />
      </RNSVGSvgView>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-start",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flexDirection": "row",
        }
      }
    >
      <Text
        accessible={true}
        align={null}
        allowFontScaling={false}
        color="rgba(0, 0, 0, 0.9)"
        ellipsizeMode="tail"
        size={48}
        style={
          Array [
            false,
            Array [
              false,
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "rgba(0, 0, 0, 0.9)",
                  "fontSize": 48,
                  "fontWeight": "700",
                },
                Array [
                  Object {
                    "lineHeight": 180,
                  },
                  Object {
                    "lineHeight": 68,
                  },
                ],
              ],
            ],
          ]
        }
        text="32"
        type={null}
        weight={700}
      >
        32
      </Text>
    </View>
    <Text
      accessible={true}
      align={null}
      allowFontScaling={false}
      color="rgba(0, 0, 0, 0.5)"
      ellipsizeMode="tail"
      size={24}
      style={
        Array [
          false,
          Array [
            false,
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "rgba(0, 0, 0, 0.5)",
                "fontSize": 24,
                "fontWeight": "400",
              },
              Array [
                Object {
                  "lineHeight": 40,
                  "marginTop": 8,
                },
                Object {
                  "lineHeight": 34,
                  "marginTop": 0,
                },
              ],
            ],
          ],
        ]
      }
      text="Current Temp"
      type={null}
      weight={400}
    >
      Current Temp
    </Text>
  </View>
</View>
`;
